mava/components/jax/updating/parameter_server.py

```diff
@@ -140,21 +128,22 @@ def on_parameter_server_init_start(self, server: SystemParameterServer) -> None:
                 server.store.parameters[
                     f"critic_{net_type_key}-{agent_net_key}"
                 ] = networks[net_type_key][agent_net_key].critic_params
-
-        # Create the checkpointer
-        if self.config.checkpoint:
-            server.store.last_checkpoint_time = 0
-
-            # Only save variables that are not empty.
-            save_variables = {}
-            for key in server.store.parameters.keys():
-                var = server.store.parameters[key]
-                # Don't store empty tuple (e.g. empty observation_network) variables
-                if not (type(var) == tuple and len(var) == 0):
-                    save_variables[key] = var
-            server.store.system_checkpointer = savers.Checkpointer(
-                save_variables, self.config.experiment_path, time_delta_minutes=0
-            )
+                if net_type_key == "networks":
+                    server.store.parameters[
+                        f"policy_opt_state-{agent_net_key}"
+                    ] = server.store.policy_opt_states[agent_net_key]
+                    server.store.parameters[
+                        f"critic_opt_state-{agent_net_key}"
+                    ] = server.store.critic_opt_states[agent_net_key]
+
+        server.store.experiment_path = self.config.experiment_path
+
+        for key, value in server.store.parameters.items():
+            if type(value) not in (list, set, dict, np.array, np.ndarray):
```

Comment:
@DriesSmit one cannot explicitly check for mutable types so I added a catchall. Lemme know your thoughts on this